market_maker_keeper/uniswapv2_util.py

```diff
@@ -0,0 +1,287 @@
+        total_eth = filter_token_df["ETH_sold"].sum()/10**18
+        return total_eth
+
+    def pool_liquidity(self, ETH_in, Token_in):
+        Token_start = self.exchange_dai_contract.functions.balanceOf(self.dai_contract_address).call()/10**18
+        ETH_start = self.web3.eth.getBalance(self.dai_contract_address)/10**18
+
+        ETH_end = (ETH_start + ETH_in) * Token_start / (Token_start + Token_in)
+        Token_end = (Token_start + Token_in) * ETH_start / (ETH_start + ETH_in)
+        ETH_out = ETH_end - ETH_start - ETH_in
+        Token_out = Token_end - Token_start - Token_in
+        return (ETH_end, Token_end, ETH_out, Token_out)
+
+    def get_future_price(self):
+
+        #getting parity pending transactions
+        a = self.web3.manager.request_blocking('parity_pendingTransactions', [])
+        #converting the pending to a pandas dataframe
+        df = pd.DataFrame(a)
+        #pruning the data frame so we are only dealing with pending transactions where the to address is to a uniswap exchange contract addess
+        # TODO: replace wit uniswap pair address
+        uni_df1 = df[df["to"].isin(self.uniswap_exch_addr)]
+
+        #checking if we have a non empty datafame
+        if len(uni_df1.index) != 0:
+            uni_df1.loc[:,'readable_input'] = uni_df1['input'].apply(self.translate_input)
+            uni_df = uni_df1[uni_df1['readable_input'] != "something wrong with input_str"]
+            bad_data_df = uni_df1[uni_df1['readable_input'] == "something wrong with input_str"]
+            if len(bad_data_df.index) != 0:
+                self.logger.info(f'----there is some bad input data for below dataframe ------ the time is {strftime("%Y-%m-%d %H:%M:%S", gmtime())}')
+                self.logger.info(bad_data_df)
+            #splitting the readable input into the functions being called vs the input numbers being sent in
+            if len(uni_df.index) != 0:
+                new = uni_df.loc[:,'readable_input'].str.split(">,", n=1, expand = True )
+                uni_df.loc[:,"Input_1st"] = new[0]
+                uni_df.loc[:,"Input_2nd"] = new[1]
+                uni_df.loc[:,'readable_gas'] = uni_df['gas'].apply(self.to_int)
+                uni_df.loc[:,'readable_gasPrice'] = uni_df['gasPrice'].apply(self.to_int)
+                uni_df.loc[:,'readable_value'] = uni_df['value'].apply(self.to_int).astype(float)
+                #printing all pending transactions that are being sent to any uniswap exchange address
+                self.logger.debug(uni_df[["to","from","readable_gas","readable_gasPrice","readable_input","Input_1st","Input_2nd","readable_value","hash"]])
+                #here we are pruning the dataframe to only pending being sent to the DAI-ETH uniswap exchange contract
+                to_df = uni_df[uni_df["to"]==self.contract_address_lower]
+
+                self.logger.info(f'total number of ETH being sent in and sold in a token to token swap '
```

Comment:
was under the impression it can cope with any pair tokens in UniswapV2 but I see logs pointing to ETH and DAI only

Reply:
That file is no longer used. After talking with Akash, we are no longer going to attempt to predict future price moves based upon pending txs. I will remove it accordingly